src/copy-paste.js

```diff
@@ -0,0 +1,347 @@
+define([
+    'jquery',
+    'underscore',
+    'vellum/mugs',
+    'vellum/tsv',
+    'vellum/core'
+], function (
+    $,
+    _,
+    mugs,
+    tsv
+) {
+    var PREAMBLE = ["Form Builder clip", "version 1"],
+        vellum,
+        offScreen = {top: -10000, left: -10000},
+        hiddenTextarea = $('<textarea></textarea>').css({
+            position: 'absolute',
+            width: 0,
+            height: 0
+        }).css(offScreen).appendTo('body');
+
+    function focusTextarea($focus, value) {
+        if ($focus.length === 0) {
+            $focus = $("body");
+        }
+        hiddenTextarea.css({top: $focus.offset().top});
+        hiddenTextarea.val(value);
+        hiddenTextarea.focus();
+        hiddenTextarea.select();
+    }
+
+    function unfocusTextarea($focus, clear) {
+        $focus.focus();
+        var value = hiddenTextarea.val();
+        if (clear && value) {
+            // HACK fix intermittent multiple-paste on Chrome (timing related?)
+            hiddenTextarea.val("");
+        }
+        return value;
+    }
+
+    function onCut(opts) {
+        var $focus = $(':focus');
+        if ($focus.is('.jstree-anchor')) {
+            var text = opts.cut();
+            if (text) {
+                focusTextarea($focus, text);
+                setTimeout(function () {
+                    unfocusTextarea($focus);
+                }, 10);
```

Comment:
Honestly I'm not sure. It's a pattern that came from @dannyroberts [original contribution of copy/paste code for Vellum](https://github.com/dimagi/Vellum/commit/05f4f59ceeee53a30845cb091bea8def4925ae77#diff-815d962c2df9702e02af486ea3984d4bR70). Handling copy/paste in JS is tricky. I'm guessing it's necessary for some strange edge case.